src/transformState.js

```diff
@@ -4,8 +4,33 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        for (const stateProper in state) {
+          if (state.hasOwnProperty(stateProper)) {
+            delete state[stateProper];
+          }
+        }
+
+        break;
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
+        break;
+      default:
+        Object.values(action.keysToRemove).forEach((removeProperties) => {
+          if (removeProperties in state) {
+            delete state[removeProperties];
+          }
+        });
+        break;
+    }
+  });
+
+  return state;
```

Comment:
Since JavaScript is pass by reference for objects, you don't need to return the state. Any modifications made to the state object will automatically affect the original state.